csharp/test/Ice/binding/AllTests.cs

```diff
@@ -238,14 +238,14 @@ public static void allTests(TestHelper helper)
                         }
                     }
 
-                    var connections = new List<Connection?>();
+                    var connections = new List<Connection>();
                     for (i = 0; i < proxies.Length; i++)
                     {
                         if (proxies[i].GetCachedConnection() != null)
                         {
-                            if (!connections.Contains(proxies[i].GetCachedConnection()))
+                            if (!connections.Contains(proxies[i].GetCachedConnection()!))
```

Comment:
It would be more elegant to hold the cached connection in a temp variable:
```
if (proxies[i].GetCachedConnection() is Connection cachedConnection)
{
  ...
}
```